### PR TITLE
Fs 2121 pull highest role for scores

### PR DESCRIPTION
**Pull request by gio-karunakaran**

### CODE REVIEW

File: app/assess/routes.py [hunk 1]
```diff
@@ -118,7 +122,9 @@ def display_sub_criteria(
         return render_template(
             "sub_criteria.html",
             on_summary=True,
-            score_list=score_list if len(score_list) > 0 else None,
+            score_list=score_list
+            if (score_list is not None and len(score_list) > 0)
+            else None,
```
Comment (tferns) [inline app/assess/routes.py:127]:
```suggestion
            score_list=score_list or None
```

File: app/assess/routes.py [hunk 2]
```diff
@@ -105,7 +105,11 @@ def display_sub_criteria(
         score_list = get_score_and_justification(
             application_id, sub_criteria_id, score_history=True
         )
-        latest_score = score_list.pop(0) if len(score_list) > 0 else None
+        latest_score = (
+            score_list.pop(0)
+            if (score_list is not None and len(score_list) > 0)
+            else None
+        )
```
Comment (tferns) [inline app/assess/routes.py:112]:
```suggestion
        latest_score = next(iter(score_list or []), None)
```

**Review by tferns — APPROVED** [review]
Looks good with some code shortening suggestions

File: app/assess/routes.py (hunk 1 above)
Comment (gio-karunakaran) [reply app/assess/routes.py:127]: 🙌🏾

File: app/assess/routes.py (hunk 2 above)
Comment (gio-karunakaran) [reply app/assess/routes.py:112]: this has introduced another error so will investigate in another ticket, for now leaving as is 👍🏾

**Pull request merged**